docs/release_notes/v0.5.0.md

```diff
@@ -0,0 +1,233 @@
+
+**Updated** Usage of Newtonsoft.Json package to System.Text.Json
+
+### Java SDK
+
+**Fixed** Hot monos in gRPC client
+
+**Added** Secrets API support
+
+**Fixed** Swallowing actor exceptions
+
+**Aded** Strongly typed actors
+
+### Samples
+
+**Fixed** Node.JS subscriber in Pub/Sub sample
+
+**Fixed** Output message for Python in Bindings sample
+
+### Docs
+
+**Added** Grafana dashboard template
+
+**Changed** `Dapr ID` to `App ID` for consistency
+
+**Fixed** Pub/Sub API reference (Thanks @lynn-orrel)
+
+**Updated** Azure Event Hubs binding spec
+
+**Refactored** Concepts and API references structure
+
+**Added** Metrics How-To tutorial
+
+**Added** Secrets API How-To tutorial
+
+**Updated** Diagrams
+
+**Added** AWS Secret Manager How-To tutorial
+
+**Added** GCP Secret Manager How-To tutorial
+
+**Added** Azure Storage Queues binding spec
+
+**Added** Azure KeyVault documentation
+
+**Added** Component authorization How-To tutorial
+
+**Added** Documentation for metrics monitoring tools
+
+**Added** Documentation for logs monitoring tools
+
+**Added** Cheat-sheet for common diagnostics queries
+
+## Upgrading to Dapr 0.5.0
+
+If you're upgrading from an older version of Dapr to 0.5.0, follow the steps here to ensure a smooth upgrade. You know, the one where you don't get red errors on the terminal.. we all hate that, right?
+
+### Local Machine / Self-hosted
+
+Uninstall Dapr using the CLI you currently have:
+
+```
+dapr uninstall --all
+```
+
+Next, get the latest CLI following [these](https://github.com/dapr/cli#installing-dapr-cli) instructions, or alternatively download the latest and greatest release from [here](https://github.com/dapr/cli/releases) and put the `dapr` binary in your PATH.
+
+Once you have downloaded the CLI, run:
+
+```
+dapr init
+```
+
+Wait for the update to finish, and you're good to go!
+Make sure you have the right runtime version (0.5.0) with:
+
+```
+dapr --version
+```
+
+### Kubernetes
+
+Download the latest CLI release as outlined above in the `Local Machine / Self-hosted` section.
+
+If you previously installed Dapr on your Kubernetes cluster using the Dapr CLI, run:
+
+```
+dapr uninstall --kubernetes
+```
+
+It's fine to ignore any errors that might show up.
+
+If you previously installed Dapr using __Helm 2.X__:
+
+```
+helm del --purge dapr
+```
+
+If you previously installed Dapr using __Helm 3.X__:
+
+```
+helm uninstall dapr -n dapr-system
+```
+
+Wait until Dapr is uninstalled, and make sure the `dapr-system` namespace is gone.
+
+If you installed Dapr with Helm to a namespace other than `dapr-system`, modify the uninstall command above to account for that.
+
+You can now follow [these](https://github.com/dapr/docs/blob/master/getting-started/environment-setup.md#using-helm-advanced) instructions on how to install Dapr using __Helm 3__.
+
+Alternatively, if you want to install Dapr in a dev/test setup, run:
+
+```
+dapr init --kubernetes
+```
+
+## Breaking Changes
```

Comment:
There is another one here, but most users won't see it.  If a user invokes daprd manually (some users have mentioned doing this), --app-id changed to --dapr-id.  This was mentioned above but not as a breaking change.

Reply:
We still don't have docs on `daprd`.. so we can't really address it here. but I'll add it to breaking changes anyway.

Reply:
Added!